capsules/src/max17205.rs

```diff
@@ -419,17 +419,23 @@ impl LegacyDriver for MAX17205Driver<'_> {
     fn subscribe(
         &self,
         subscribe_num: usize,
-        callback: Option<Callback>,
+        callback: Callback,
         _app_id: AppId,
-    ) -> ReturnCode {
+    ) -> Result<Callback, (Callback, ErrorCode)> {
         match subscribe_num {
             0 => {
-                self.callback.insert(callback);
-                ReturnCode::SUCCESS
+                if let Some(prev) = self.callback.replace(callback) {
+                    Ok(prev)
+                } else {
+                    // TODO(alevy): This should never happen because we start with a full MapCell
+                    // and only ever replace it. This is just defensive until this module becomes
+                    // multi-user, which will preclude the need for a MapCell in the first place.
+                    Ok(Callback::default())
+                }
```

Comment:
Given we initialize the `MapCell` with a `Callback` and only ever swap one for another, this case should -- as you state -- never happen. I would've expected to just panic (`.expect`) instead of failing silently, as something is clearly off then. But I don't have any strong opinion about it, and this is a fair documentation placeholder :smile:.

Reply:
Having the fallback case seemed more resilient to me. In particular, I can imagine someone might come back to this capsule at some point and notice that the `MapCell` is initialized to some strange default value rather than `empty` the way it's used everywhere else. If they decide to change that without carefully reasoning about the particular semantics of this capsule, and none of us catch it... everything will still work exactly as intended because of this fallback case.